src/plugins/explore/public/components/visualizations/table/table_vis.tsx

```diff
@@ -3,56 +3,264 @@
-    (perPage) =>
-      setPagination((p) => ({
-        ...p,
-        pageSize: perPage,
-        pageIndex: 0,
-      })),
-    [setPagination]
+  const onChangeItemsPerPage = useCallback((newPageSize: number) => {
+    setPagination((p) => ({
+      ...p,
+      pageSize: newPageSize,
+      pageIndex: 0,
+    }));
+  }, []);
+
+  useEffect(() => {
+    if (!styleOptions?.showColumnFilter) {
+      setFilters({});
+    }
+  }, [styleOptions?.showColumnFilter]);
+
+  const filteredRows = useMemo(() => {
+    return rows.filter((row) =>
+      Object.entries(filters).every(([columnId, config]) => {
+        const v = row[columnId];
+        return matchesFilter(v, config);
+      })
+    );
+  }, [rows, filters]);
+
+  useEffect(() => {
+    setPagination({ pageIndex: 0, pageSize });
+  }, [filteredRows.length, pageSize]);
+
+  const availableFieldSet = useMemo(() => new Set(columns.map((c) => c.column)), [columns]);
+  const normalizedFooterCalcs = useMemo<Calc[]>(() => {
+    const raw = styleOptions?.footerCalculations ?? [];
+
+    const processed: Calc[] = (raw as any[]).map(
+      (c): Calc =>
+        'fields' in c
+          ? {
+              fields: (c.fields ?? [])
+                .filter((f: string | number | null | undefined): f is string | number => f != null)
+                .map(String),
+              calculation: c.calculation as CalculationMethod,
+            }
+          : {
+              fields: c.field != null ? [String(c.field)] : [],
+              calculation: c.calculation as CalculationMethod,
+            }
+    );
+
+    return processed
+      .map((c) => ({
+        ...c,
+        fields: c.fields.filter((f) => availableFieldSet.has(f)),
+      }))
+      .filter((c) => c.fields.length > 0);
+  }, [styleOptions?.footerCalculations, availableFieldSet]);
+
+  const footerValues = useMemo(() => {
+    if (!styleOptions?.showFooter || !normalizedFooterCalcs.length) return undefined;
+
+    const footer: Record<string, any> = {};
+
+    const getNumericSeries = (field: string) =>
+      filteredRows.map((row) => row[field]).filter((v) => typeof v === 'number' && !isNaN(v));
+
+    normalizedFooterCalcs.forEach(({ fields, calculation }) => {
+      fields.forEach((field) => {
+        const values = getNumericSeries(field);
+        if (values.length === 0) {
+          footer[field] = '-';
+          return;
+        }
+
+        const result = calculateValue(values, calculation);
+        if (result == null) {
+          footer[field] = '-';
+        } else {
+          const label = calculation.charAt(0).toUpperCase() + calculation.slice(1);
+          footer[field] = `${label}: ${result}`;
+        }
+      });
+    });
+    return footer;
+  }, [filteredRows, styleOptions?.showFooter, normalizedFooterCalcs]);
+
+  const getCellColor = useCallback(
+    (value: any, columnId: string): string | undefined => {
+      if (columnTypes[columnId] !== 'numerical' || !styleOptions?.thresholds) return undefined;
```

Comment:
I think we don't need to check `columnTypes[columnId] !== 'numerical'` here, because it does convert the value to number later `const numValue = Number(value);`. 

This will make the implementation more resilient, for example, for status code column, string value `"400"`, `"500"` can also be colored by thresholds.